### PR TITLE
topics tracklist

### PR DESCRIPTION
**Pull request by carlynlee**
shortens the tracklist, Match color of tracklisting with schedule

result of these changes: 
Screenshot 2024-03-04 at 3 31 16 PM

### CODE REVIEW

File: server/style.css [hunk 1]
```diff
@@ -466,237 +466,102 @@ font {
...
-  background: #956140 !important;
-  color: #fff !important;
+  background-color: #956140;
+  color: #FFF;
 }
-
-.Youth {
-  background: #098C09 !important;
-  color: #fff !important;
+.FOSS@HOME {
```
Comment (kylerisse) [inline server/style.css:561]: I'm concerned the use of `@` in the selector name might not be valid. Can you confirm @carlynlee @sarcasticadmin ?

**Review by kylerisse — COMMENTED** [review]
Thanks @carlynlee . Appreciate you updating the color schemes for new and renamed tracks. I just have one concern noted in line. Generally not a fan of special characters in map keys and unsure how that might affect css.

File: server/style.css (hunk 1 above)
Comment (sarcasticadmin) [reply server/style.css:561]: thats a good question, let me do some digging

Comment (sarcasticadmin) [reply server/style.css:561]: From the example that @carlynlee sent over while work through this it does look like `FLOSS@HOME` is not the expected color. Its grey (default) in this screenshot:

Screenshot 2024-03-04 at 3 31 16 PM

I believe we just need to escape the `@`:

```
.FOSS\@HOME {
```

If you want a quick example in https://jsfiddle.net/ :

html
```
<div class="container">
  <div class="one">
    <p>One</p>
  </div>
  <div class="floss@home">
    <p>floss@home</p> 
  </div>
  <div class="floss@broken">
    <p>floss@broken</p>
  </div>
</div>
```

css:
```
/* For each class, set some colors */
.one {
  background-color: cornflowerblue;
}
.floss\@home {
  color: black;
  background-color: #8ADA20;
}
.floss@broken {
  color: black;
  background-color: #8ADA20;
}
```

Results in:
ss-202403051709612052

File: README.md [hunk 2]
```diff
@@ -86,6 +86,7 @@ There is a bit of manual effort necessary from year to year. These tasks include
 * update `$shorten_topics` in `scroll.php` to reflect the updated track list, matching exactly the keys to what is being supplied by the xml from drupal minus spaces
 * update `$shorten_topics` in `room.php` to match `scroll.php`
 * create a style for each of the keys in `$shorten_topics` in `style.css` matching it to the public site
+* add colors for topics/tracks in `style.css`
```
Comment (sarcasticadmin) [inline README.md:89]: Thanks for adding this to the list!

**Review by sarcasticadmin — COMMENTED** [review]
@carlynlee topics are looking good and awesome job getting these colors to align with the [main schedule](https://www.socallinuxexpo.org/scale/21x/schedule/thursday) ! This will really go a along way to make it easy for the Scale attendees to identify their designed tracks since it will match the program.

Per Kyles suggestion, I think we just need to make that small tweak in the `css` (see my comment inline below) and then it should be good. Would you mind testing that out?

File: server/style.css (hunk 1 above)
Comment (kylerisse) [reply server/style.css:561]: I believe the selector name is the map key value. My suggestion is to just omit the `@` from the key here https://github.com/socallinuxexpo/scale-signs/blob/e166012f2f5f74fee75d16be98c3edaab9d4163a/server/scroll.php#L50 , making it FOSSHOME` and then match the CSS selector to `FOSSHOME`. Same pattern as how spaces are treated. Probably cleaner than escaping.

Comment (sarcasticadmin) [reply server/style.css:561]: that makes sense, definitely safer :+1:

File: README.md (hunk 2 above)
Comment (carlynlee) [reply README.md:89]: Ok, I removed the special char @ from two files:

https://github.com/carlynlee/scale-signs/blob/shorten_topics_tracklist/server/scroll.php#L50

https://github.com/carlynlee/scale-signs/blob/shorten_topics_tracklist/server/style.css#L561

I rebuilt the image and tried to view the track listing again but I'm unsure why the colors aren't applied:
Screenshot 2024-03-04 at 3 31 16 PM

Comment (sarcasticadmin) [reply README.md:89]: did you rebuild the container? It looks like its still the old name `Foss @ Home` from your screenshot (which it should be `FOSS HOME` from your diff)

BTW keep the `@` on the right side of the assignment. So the entry in `scroll.php` would look like:

```
"FOSSHOME"	=>	 "FOSS @ HOME",
```

Comment (carlynlee) [reply README.md:89]: yep, I ran `docker-compose build --no-cache` and then `docker-compose up -d`

File: server/scroll.php [hunk 3]
```diff
@@ -69,9 +75,10 @@
   $node->{'Time'} = preg_replace('/<[^>]*>/', '', $node->{'Time'});
   $node->{'Day'} = preg_replace('/<[^>]*>/', '', $node->{'Day'});
   
-  // Remove Spaces so we can use it for a CSS class
+  // Remove special chars in Topic from XML request so we can use it for a CSS class
   $node->{'Topic'} = preg_replace('/\s+/', '', $node->{'Topic'});
   $node->{'Topic'} = preg_replace('/\&/', 'and', $node->{'Topic'});
+  $node->{'Topic'} = preg_replace('/\@/', '', $node->{'Topic'});
```
Comment (sarcasticadmin) [inline server/scroll.php:81]: @carlynlee did a little digging, turns out we need to replace the `@` in the original Topic name we get from the XML as well. Its due to the lookup that takes place here: https://github.com/socallinuxexpo/scale-signs/blob/2c23457c3f9656ac391a9be722490ba4d287c683/server/scroll.php#L257

It all looks good for me now :slightly_smiling_face: 
ss-202403061709684396

I pushed these changes directly on your branch so just pull them down with: `git pull -r origin shorten_topics_tracklist` and you should these my additions

Comment (carlynlee) [reply server/scroll.php:81]: Thanks @sarcasticadmin ! I pulled your changes and it seems to work on my end too:
Screenshot 2024-03-06 at 9 43 21 AM

**Review by sarcasticadmin — APPROVED** [review]
@carlynlee this is fantastic, the signs server is really coming along :tada: 

I went ahead and `rebased` these commits so the history was more straightforward prior to use merging it into `master`

**Pull request merged**